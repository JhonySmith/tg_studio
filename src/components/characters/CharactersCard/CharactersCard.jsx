import { Gender } from 'components/Gender/Gender.jsx';
import { useState } from 'react';
import { CharacterDetails } from '../CharacterDetails/CharacterDetails.jsx';
import * as S from './CharactersCard.styles.js';

export const CharactersCard = (props) => {
  const { character } = props;
  const { name, mass, height, gender } = character;

  const [isDetailsOpen, setIsDetailsOpen] = useState(false);

  const openDetailsHandler = () => {
    setIsDetailsOpen(!isDetailsOpen);
  };

  const closeDetailsHandler = () => {
    setIsDetailsOpen(false);
  };

  return (
    <S.Card onClick={openDetailsHandler} role="card">
      {isDetailsOpen ? (
        <CharacterDetails
          character={character}
          closeHandler={closeDetailsHandler}
        />
      ) : null}
      <S.Name>{name}</S.Name>
      <S.RounParamBlock>
        <S.RounParamWraper>
          <S.RoundParam>{height === 'unknown' ? 'N/A' : height}</S.RoundParam>
          <S.RoundParamLabel>height</S.RoundParamLabel>
        </S.RounParamWraper>
        <S.RounParamWraper>
          <S.RoundParam>{mass === 'unknown' ? 'N/A' : mass}</S.RoundParam>
          <S.RoundParamLabel>mass</S.RoundParamLabel>
        </S.RounParamWraper>
      </S.RounParamBlock>
      <S.FooterParams>
        <Gender gender={gender} />
      </S.FooterParams>
    </S.Card>
  );
};
