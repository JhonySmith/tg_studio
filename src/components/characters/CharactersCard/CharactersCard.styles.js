import styled from 'styled-components';
import { karlaFont } from 'styles/variables/fonts';

export const Card = styled.li`
  box-sizing: border-box;

  width: 351.9px;
  padding: 6px 26px;

  background: #f0f0f0;
  border-radius: 8px;

  box-shadow: 4px 4px 4px rgba(33, 33, 33, 0.1);
`;

export const Name = styled.p`
  font-family: ${karlaFont};
  font-style: normal;
  font-weight: 700;
  font-size: 18px;
  line-height: 21px;
`;

export const RounParamBlock = styled.div`
  display: flex;
`;

export const RounParamWraper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  margin-right: 13px;
`;

export const RoundParam = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  width: 37.95px;
  height: 36.04px;

  border: 3px solid #212121;
  border-radius: 50%;
`;

export const RoundParamLabel = styled.span`
  margin-top: 5px;

  font-family: ${karlaFont};
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 14px;
  text-align: center;

  color: #4b4b4b;
`;

export const ColorBlock = styled.div`
  width: fit-content;
  padding: 2px 10px;

  box-shadow: inset 0px -2px 0px rgba(0, 0, 0, 0.18);
  border-radius: 11px;

  font-family: 'Roboto';
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 14px;
  text-align: center;

  color: #212121;
`;

export const Male = styled(ColorBlock)`
  background: #73d677;
`;

export const Female = styled(ColorBlock)`
  background: #c956ff;
`;

export const Hermaphrodite = styled(ColorBlock)`
  background: #f5db13;
`;

export const FooterParams = styled.div`
  margin-top: 10px;
`;
