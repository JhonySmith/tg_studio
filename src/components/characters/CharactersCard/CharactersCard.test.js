import { CharactersCard } from './CharactersCard.jsx';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

const mockCharacter = {
  name: 'Some name',
  mass: '123',
  height: '134',
  gender: 'male',
};

describe('Test character card', () => {
  test('Snap test', () => {
    render(<CharactersCard character={mockCharacter} />);

    const tree = screen.getByRole('card');
    expect(tree).toMatchSnapshot();
  });

  test('Element in DOM', () => {
    render(<CharactersCard character={mockCharacter} />);
    const tree = screen.getByRole('card');
    expect(tree).toBeInTheDocument();
  });

  test('Right data test', () => {
    render(<CharactersCard character={mockCharacter} />);

    const name = screen.getByText('Some name');
    const mass = screen.getByText('123');
    const height = screen.getByText('134');
    const gender = screen.getByText('male');

    expect(name).toBeInTheDocument();
    expect(mass).toBeInTheDocument();
    expect(height).toBeInTheDocument();
    expect(gender).toBeInTheDocument();
  });

  test('Click card test', () => {
    render(<CharactersCard character={mockCharacter} />);

    const card = screen.getByRole('card');
    fireEvent.click(card);

    const details = screen.getByRole('details');
    expect(details).toBeInTheDocument();
  });
});
