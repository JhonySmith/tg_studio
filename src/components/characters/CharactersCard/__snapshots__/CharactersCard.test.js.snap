// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test card 1`] = `
<li
  class="sc-fEXmlR hhtYUo"
  role="card"
>
  <p
    class="sc-bjfHbI BzANE"
  >
    Some name
  </p>
  <div
    class="sc-idXgbr dwFavs"
  >
    <div
      class="sc-dIfARi hKAUbf"
    >
      <div
        class="sc-hHTYSt gjmgcj"
      >
        134
      </div>
      <span
        class="sc-dmctIk jvgna"
      >
        height
      </span>
    </div>
    <div
      class="sc-dIfARi hKAUbf"
    >
      <div
        class="sc-hHTYSt gjmgcj"
      >
        123
      </div>
      <span
        class="sc-dmctIk jvgna"
      >
        mass
      </span>
    </div>
  </div>
  <div
    class="sc-cwSeag jMzkyS"
  >
    <div
      class="sc-bcXHqe sc-gswNZR kbRfMT dOSFAp"
    >
      male
    </div>
  </div>
</li>
`;

exports[`Test character card Element in DOM 1`] = `
<li
  class="sc-fEXmlR hhtYUo"
  role="card"
>
  <p
    class="sc-bjfHbI BzANE"
  >
    Some name
  </p>
  <div
    class="sc-idXgbr dwFavs"
  >
    <div
      class="sc-dIfARi hKAUbf"
    >
      <div
        class="sc-hHTYSt gjmgcj"
      >
        134
      </div>
      <span
        class="sc-dmctIk jvgna"
      >
        height
      </span>
    </div>
    <div
      class="sc-dIfARi hKAUbf"
    >
      <div
        class="sc-hHTYSt gjmgcj"
      >
        123
      </div>
      <span
        class="sc-dmctIk jvgna"
      >
        mass
      </span>
    </div>
  </div>
  <div
    class="sc-cwSeag jMzkyS"
  >
    <div
      class="sc-bcXHqe sc-gswNZR kbRfMT dOSFAp"
    >
      male
    </div>
  </div>
</li>
`;

exports[`Test character card Right data test 1`] = `
<li
  class="sc-fEXmlR hhtYUo"
  role="card"
>
  <p
    class="sc-bjfHbI BzANE"
  >
    Some name
  </p>
  <div
    class="sc-idXgbr dwFavs"
  >
    <div
      class="sc-dIfARi hKAUbf"
    >
      <div
        class="sc-hHTYSt gjmgcj"
      >
        134
      </div>
      <span
        class="sc-dmctIk jvgna"
      >
        height
      </span>
    </div>
    <div
      class="sc-dIfARi hKAUbf"
    >
      <div
        class="sc-hHTYSt gjmgcj"
      >
        123
      </div>
      <span
        class="sc-dmctIk jvgna"
      >
        mass
      </span>
    </div>
  </div>
  <div
    class="sc-cwSeag jMzkyS"
  >
    <div
      class="sc-bcXHqe sc-gswNZR kbRfMT dOSFAp"
    >
      male
    </div>
  </div>
</li>
`;

exports[`Test character card Snap test 1`] = `
<li
  class="sc-fEXmlR hhtYUo"
  role="card"
>
  <p
    class="sc-bjfHbI BzANE"
  >
    Some name
  </p>
  <div
    class="sc-idXgbr dwFavs"
  >
    <div
      class="sc-dIfARi hKAUbf"
    >
      <div
        class="sc-hHTYSt gjmgcj"
      >
        134
      </div>
      <span
        class="sc-dmctIk jvgna"
      >
        height
      </span>
    </div>
    <div
      class="sc-dIfARi hKAUbf"
    >
      <div
        class="sc-hHTYSt gjmgcj"
      >
        123
      </div>
      <span
        class="sc-dmctIk jvgna"
      >
        mass
      </span>
    </div>
  </div>
  <div
    class="sc-cwSeag jMzkyS"
  >
    <div
      class="sc-bcXHqe sc-gswNZR kbRfMT dOSFAp"
    >
      male
    </div>
  </div>
</li>
`;
